src/store/sagas/dataset-list-saga.js

```diff
@@ -1,15 +1,15 @@
-import { takeEvery } from 'redux-saga/effects'
-import { RETRIEVE_DATA_LIST, selectDataList, displayError } from '../actions'
+import { takeEvery, select, take } from 'redux-saga/effects'
+import { DATASET_SEARCH, UPDATE_DATASET_SEARCH_PARAMS, datasetSearchSucceeded, displayError } from '../actions'
 import apiInvoker from './api-invoker'
+import { getSearchParams } from '../selectors'
+
 
 export default function * theRealDatasetSaga () {
-  yield takeEvery(RETRIEVE_DATA_LIST, apiInvoker({ endpoint: '/api/v1/dataset/search', actionator: selectDataList, queryParameterBuilder }))
+  while (true) {
+    yield take(UPDATE_DATASET_SEARCH_PARAMS)
+    let search = yield take(DATASET_SEARCH)
+    const params = yield select(getSearchParams);
```

Comment:
Why can't se get these params from the UPDATE_DATASET_SEARCH_PARAMS action?

Reply:
That might be a good step we can take now?

Reply:
UPDATE_DATASET_SEARCH_PARAMS doesn't get sent the whole picture every time. It just takes a patch to the params, essentially.